pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -210,6 +200,13 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard, ICo
 
         _setManagementAumFeePercentage(params.managementAumFeePercentage);
 
+        // Write the scaling factors for each token into their token state.
+        // We do this before setting the weights in `_startGradualWeightChange` so we start from a empty token state.
+        for (uint256 i = 0; i < totalTokens; i++) {
+            IERC20 token = params.tokens[i];
+            _tokenState[token] = ManagedPoolTokenLib.setTokenScalingFactor(bytes32(0), token);
```

Comment:
Bit odd that we don't use the initialization function here

Reply:
Will address separately.